src/plan.rs

```diff
@@ -502,3 +502,355 @@ impl Assets {
         self.absolute_timelock = b.absolute_timelock.or(self.absolute_timelock);
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::str::FromStr;
+
+    use bitcoin::{LockTime, Sequence};
+
+    use super::*;
+    use crate::*;
+
+    fn test_inner(
+        desc: &str,
+        keys: Vec<DescriptorPublicKey>,
+        hashes: Vec<hash160::Hash>,
+        // [ (key_indexes, hash_indexes, older, after, expected) ]
+        tests: Vec<(
+            Vec<usize>,
+            Vec<usize>,
+            Option<Sequence>,
+            Option<LockTime>,
+            Option<usize>,
+        )>,
+    ) {
+        let desc = Descriptor::<DefiniteDescriptorKey>::from_str(&desc).unwrap();
+
+        for (key_indexes, hash_indexes, older, after, expected) in tests {
+            let mut assets = Assets::new();
+            if let Some(seq) = older {
+                assets = assets.older(seq);
+            }
+            if let Some(locktime) = after {
+                assets = assets.after(locktime);
+            }
+            for ki in key_indexes {
+                assets = assets.add(keys[ki].clone());
+            }
+            for hi in hash_indexes {
+                assets = assets.add(hashes[hi].clone());
+            }
+
+            let result = desc.get_plan(&assets);
+            assert_eq!(
+                result.as_ref().map(|plan| plan.satisfaction_weight()),
+                expected,
+                "{:#?}",
+                result
+            );
+        }
+    }
+
+    #[test]
+    fn test_or() {
```

Comment:
Either we can have key[0] or key[1] so for `test_or` the formula is 
`4 (scriptSig len) + 1 (witness len) + 73 (sig)`